### PR TITLE
Issue #1385 Attach should not reject changes

### PR DESCRIPTION
**Pull request by lukewaters**
This adds a code path to attach entities and update saved values to the current values

### CODE REVIEW

**Comment by lukewaters** [comment]
@ajcvickers

File: src/EntityFramework.Core/ChangeTracking/EntityEntry.cs [hunk 1]
```diff
@@ -32,6 +32,13 @@ public virtual EntityState State
             }
         }
 
+        public virtual void AttachState(EntityState value)
```
Comment (ajcvickers) [inline src/EntityFramework.Core/ChangeTracking/EntityEntry.cs]: What is this method for?

File: src/EntityFramework.Core/DbContext.cs [hunk 2]
```diff
@@ -511,6 +511,16 @@ private EntityEntry EntryWithoutDetectChanges([NotNull] object entity)
             return entry;
         }
 
+        private EntityEntry<TEntity> SetEntityStateAttached<TEntity>(TEntity entity, EntityState entityState) where TEntity : class
+        {
+            var entry = EntryWithoutDetectChanges(entity);
+
+            entry.AttachState(entityState);
+
+            return entry;
```
Comment (ajcvickers) [inline src/EntityFramework.Core/DbContext.cs]: I don't think it should be done this way. We don't want to add new surface to EntityEntry for this. Instead, get the StateEntry itself, set the state to Unchanged passing in true for the acceptChanges flag. Then create an EntityEntry around that StateEntry and return it.

Comment (ajcvickers) [inline src/EntityFramework.Core/DbContext.cs]: Also, make sure that all the Attach and AttachRange methods on DbContext and DbSet have this behavior, and add tests to ensure this.

File: src/EntityFramework.Core/ChangeTracking/EntityEntry.cs (hunk 1 above)
Comment (lukewaters) [inline src/EntityFramework.Core/ChangeTracking/EntityEntry.cs]: Removed

**Comment by ajcvickers** [comment]
:shipit:

**Comment by whyi** [comment]
:shipit:

**Pull request merged**